action-types.yml

```diff
@@ -0,0 +1,35 @@
+typingSpec: krzema12/github-actions-typing@v0.3
+inputs:
+  distribution:
+    type: enum
+    allowedValues:
+     - 'Alpine'
+     - 'Debian'
+     - 'kali-linux'
+     - 'openSUSE-Leap-15.2'
+     - 'Ubuntu-20.04'
+     - 'Ubuntu-18.04'
+     - 'Ubuntu-16.04'
+  use-cache:
+    type: boolean
+  wsl-conf:
+    type: string
+  set-as-default:
+    type: boolean
+  update:
+    type: boolean
+  additional-packages:
+    type: list
+    separator: ' '
+    listItem:
+      type: string
```

Comment:
Yes, there can be more, e.g. for an enum (see [here](https://github.com/krzema12/github-actions-typing/#list)) or an integer with named values. In terms of typing, whatever goes under `listItem` behaves similarly like action's input/output. The only limitation for now is that list items cannot themselves be lists - I want to keep it simple for now. No use case for `List<List<List<String>>>` appeared yet :wink:

Reply:
No use case for `List<Boolean>` either :)

Reply:
Yeah, we could simplify it further and allow lists of only integers and enums - I wanted to do it initially. However, then I asked myself what if we allow basic types there, and forbid only nesting lists because then it may become more tricky. So allowing e.g. `List<Boolean>` stems from the idea of supporting lists of basic types, not having a real-life use case for it :) Would you guys recommend limiting lists only to lists of strings and enums? We can always extend it later.

Reply:
FTR, this discussion doesn't seem to be relevant for the decision if there should be `list-item:` with subkeys or `list-item-type:` because the enum kinda requires the former.

Reply:
`list-item` feels better to me

Reply:
Ah, right, yeah, if you need more subkeys potentially like for enums, `list-item` is fine of course.
I think I'd not limit the subtype (further than forbidding lists unless someone comes up with a use-case).
If someone wants a list of booleans, why not, if you don't need special code to support it and I guess you don't , or at least not significant.